flow/declarations/mobx.js

```diff
@@ -262,15 +262,8 @@ declare module 'mobx' {
     // observable overloads
     (target: Object, key: string, baseDescriptor?: PropertyDescriptor): any,
     <T>(value: Array<T>): IObservableArray<T>,
-    (value: string): IObservableValue<string>,
-    (value: boolean): IObservableValue<boolean>,
-    (value: number): IObservableValue<number>,
-    (value: Date): IObservableValue<Date>,
-    (value: RegExp): IObservableValue<RegExp>,
-    (value: Function): IObservableValue<Function>,
     <T>(value: null | void): IObservableValue<T>,
     (value: null | void): IObservableValue<any>,
-    (): IObservableValue<any>,
```

Comment:
Explanation: this took priority over the more generic
`<T: Object>(value: T): T,`
so I delete these so it infers the type properly. The reason we didn't do this before is that when this code was written, Flow's support for generics wasn't good enough to handle it (but now it is!)